Copyright (c) 2004, Jan Kneschke, incremental
All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:

- Redistributions of source code must retain the above copyright notice, this
list of conditions and the following disclaimer.

- Redistributions in binary form must reproduce the above copyright notice,
this list of conditions and the following disclaimer in the documentation
and/or other materials provided with the distribution.

- Neither the name of the 'incremental' nor the names of its contributors may
be used to endorse or promote products derived from this software without
specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
THE POSSIBILITY OF SUCH DAMAGE.

Date: Aug 31, 2016 12:46 PM

From: e4471ee03f0ed43bb17107373556e45a2cb0d129 Sep 15, 2016 1:52 AM

Subject: [PATCH] RDKB-6774,RDKB-6775,RDKB-7628: lighttpd to 1.4.39

	RDKB-6774 upgrading lighttpd to 1.4.39, RDKB-6775 mod_csrf changes for lighttpd_1.4.39.

Source: COMCAST

Upstream-Status: Pending

Signed-off-by: rthangamani <ramakrishnan.thangamani@wipro.com>
---

Index: lighttpd-1.4.39/src/network.c
===================================================================
--- lighttpd-1.4.39.orig/src/network.c
+++ lighttpd-1.4.39/src/network.c
@@ -273,6 +273,20 @@
 		goto error_free_socket;
 	}
 
+#ifdef SO_BINDTODEVICE
+        /* host might be an interface name, so try to bind to it */
+        if (host && (srv_socket->addr.plain.sa_family == AF_INET
+#ifdef HAVE_IPV6
+                     || srv_socket->addr.plain.sa_family == AF_INET6)) {
+#else
+                    )) {
+#endif
+                if (!setsockopt(srv_socket->fd, SOL_SOCKET, SO_BINDTODEVICE,
+                                host, strlen(host)))
+                        host = NULL;
+        }
+#endif
+
 	switch(srv_socket->addr.plain.sa_family) {
 #ifdef HAVE_IPV6
 	case AF_INET6:
